docs/cloud-edge/modules/user-agent-filter.md

```diff
@@ -0,0 +1,135 @@
+# User Agent Filter
+
+---
+
+## Overview
+
+User Agent Filter module allows you to allow or deny traffic based on the user-agent request header. You can define allow and deny regular expression rules that apply to the incoming user-agent request header.
+
+## Quickstart
+
+### Agent CLI
+
+```bash
+ngrok http 80 --ua-filter-allow="(foo/(\d)+.(\d)+)","(bar/(\d)+.(\d)+)" --ua-filter-deny="(buz/(\d)+.(\d)+)"
+```
+
+### SSH
+
+```bash
+ssh -R 443:localhost:80 connect.ngrok-agent.com http \
+ --ua-filter-allow="(foo/(\d)+.(\d)+)" \
+ --ua-filter-deny="(bar/(\d)+.(\d)+)" \
+```
+
+### Agent Configuration File
+
+```yaml
+tunnels:
+ example:
+   proto: https
+   addr: 80
+   ua_filter_allow: ["(foo/(\d)+.(\d)+)", "(bar/(\d)+.(\d)+)"]
+   ua_filter_deny: ["(buzz/(\d)+.(\d)+)"]
+```
+
+### Go SDK
+
+```go
+import (
+	"context"
+	"net"
+
+	"golang.ngrok.com/ngrok"
+	"golang.ngrok.com/ngrok/config"
+)
+
+func listenUserAgentFIlter(ctx context.Context) net.Listener {
+	listener, _ := ngrok.Listen(ctx,
+		config.HTTPEndpoint(
+			config.WithUserAgentFilter( []string{"(foo)/(\d)+.(\d)+"}, []string{"bar/(\d)+.(\d)+"}),
+		),
+		ngrok.WithAuthtokenFromEnv(),
+	)
+	return listener
+}
+```
+
+## Behavior
+
+### Rule Evaluation
+
+On request, the User Agent Filter module will check the incoming `user-agent` header value against the list of defined allow and deny regular expression rules.
+
+Requests with no `user-agent` header will be denied by deafult. If you wish to accept these you can add a allow rule for empty strings: `^$`
+
+Requests with multiple `user-agent` headers will only match the first User-Agent header.
```

Comment:
Is there a distinction between `user-agent` and User-Agent? I feel like "Requests with multiple `user-agent` headers will only match the first header." is clear enough.